tensorflow/python/ops/resources.py

```diff
@@ -86,7 +87,8 @@ def report_uninitialized_resources(resource_list=None,
     resource_list = shared_resources() + local_resources()
   with ops.name_scope(name):
     # Run all operations on CPU
-    with ops.device("/cpu:0"):
+    local_device = os.environ.get("TF_LOCAL_DEVICE", "/cpu:0")
```

Comment:
TF_LOCAL_DEVICE seems a generic name. It may collide with some existing users. 
Is it possible to find a more specific name, something like 'TF_DEVICE_REPORT_UNINITIALIZED'.

Could you please also give us info about whether you have run this and got problem solved?

Reply:
Here is the issue. https://github.com/tensorflow/tensorflow/issues/22137

Reply:
So I assume this PR will fix that issue.

Reply:
Yes.  "with ops.device('/cpu:0')" also appears at some other places, such as 'tensorflow/python/estimator/estimator.py'. I think it will lead to OOM when the workers are too many, but I do not use it in my case.

Reply:
@ispirmustafa TF_DEVICE_REPORT_UNINITIALIZED seems too special for users and they may not know what it means.

Reply:
TF_LOCAL_DEVICE is very generic, and I don't think users will know what it means either.

In particular, since this is not documented, nobody will know what this does without looking through the code. I'm not sure we necessarily want to widely broadcast this, but I agree with @ispirmustafa that a more descriptive name for the environment variable would be better. Maybe TF_DEVICE_FOR_UNINITIALIZED_VARIABLE_REPORTING? Since that's all it's used for.

Reply:
@martinwicke @ispirmustafa I agree with you in some way. The root cause of this bug is device setting. When only '/cpu:0' is assigned, the runtime will pick the first job name and task index in alphabet order to fill the device name. That is why '/job:ps/task:0/cpu:0' was used when running in ps-worker mode. I can not evaluate what is the impact when the default device placement is changed. So I just fix the problem case by case.